src/sentry/api/event_search.py

```diff
@@ -1056,63 +1104,84 @@ def to_list(value):
 
 
 def format_search_filter(term, params):
-    project_to_filter = None  # Used to avoid doing multiple conditions on project ID
+    projects_to_filter = []  # Used to avoid doing multiple conditions on project ID
     conditions = []
     group_ids = None
     name = term.key.name
     value = term.value.value
     if name in (PROJECT_ALIAS, PROJECT_NAME_ALIAS):
         if term.operator == "=" and value == "":
             raise InvalidSearchQuery("Invalid query for 'has' search: 'project' cannot be empty.")
-        project = None
-        try:
-            project = Project.objects.get(id__in=params.get("project_id", []), slug=value)
-        except Exception as e:
-            if not isinstance(e, Project.DoesNotExist) or term.operator != "!=":
-                raise InvalidSearchQuery(
-                    f"Invalid query. Project {value} does not exist or is not an actively selected project."
-                )
-        else:
+        slugs = to_list(value)
+        projects = {
+            p.slug: p.id
+            for p in Project.objects.filter(id__in=params.get("project_id", []), slug__in=slugs)
+        }
+        missing = [slug for slug in slugs if slug not in projects]
+        if missing and term.operator not in ("!=", "NOT IN"):
+            raise InvalidSearchQuery(
+                f"Invalid query. Projects {missing} do not exist or are not actively selected."
+            )
+        project_ids = list(sorted(projects.values()))
+        if project_ids:
             # Create a new search filter with the correct values
-            term = SearchFilter(SearchKey("project_id"), term.operator, SearchValue(project.id))
+            term = SearchFilter(
+                SearchKey("project_id"),
+                term.operator,
+                SearchValue(project_ids if term.is_in_filter else project_ids[0]),
+            )
             converted_filter = convert_search_filter_to_snuba_query(term)
             if converted_filter:
-                if term.operator == "=":
-                    project_to_filter = project.id
-
+                if term.operator in equality_operators:
+                    projects_to_filter = project_ids
                 conditions.append(converted_filter)
     elif name == ISSUE_ID_ALIAS and value != "":
         # A blank term value means that this is a has filter
         group_ids = to_list(value)
     elif name == ISSUE_ALIAS:
         operator = term.operator
-        if value == "unknown":
-            # `unknown` is a special value for when there is no issue associated with the event
-            operator = "=" if term.operator == "=" else "!="
-            value = ""
-        elif value != "" and params and "organization_id" in params:
+        value = to_list(value)
+        group_short_ids = [v for v in value if v and v != "unknown"]
```

Comment:
Could we keep the unknown comment here?
```suggestion
        # `unknown` is a special value for when there is no issue associated with the event
        group_short_ids = [v for v in value if v and v != "unknown"]
```

Also these variable names make me a bit nervous since we're re-using `value` a bunch, could we change 'em so there isn't any overlap?